### ISSUE TITLE
within_iframe needs a name/id

### ISSUE THREAD

**Issue by PragTob** [teampoltergeist/poltergeist#630]
Hello and thank you for your good work :)

I'm in the situation that I want to test interaction with an iframe that is not completely under my control and that has neither an id nor a name (I create a div with an id, and some 3rd party JavaScript puts the iframe inside the div).

That works when I run with chrome, but not when I run with poltergeist (1.6.0) - sadly. By the looks of it only iframes with names/ids are supported: https://github.com/teampoltergeist/poltergeist/blob/master/lib/capybara/poltergeist/browser.rb#L125-L130

Is there a way to make it work just given a capybara node? Would be happy to help on this.

e.g. current code that works with chrome but not poltergeist is:

```
iframe = find('#id-of-the-div iframe')
  within_frame iframe do
    # MAGIC
  end
```

As a workaround I was thinking about adding a name/id to the iframe myself but it seems that doesn't work as well due to #559 

Thanks!
Tobi

**Comment by Hainish** [comment]
We're also encountering this for our project at https://github.com/EFForg/phantom-of-the-capitol, would love to see fixed

**Comment by mamhoff** [comment]
I had the same issue, and found an (admittedly not very nice) workaround: The iframes can be referenced using an integer ID, and the library I was interacting was nice enough to always add the iframes in the same order. That way I could do the following: 
```
  within_frame(0) { fill_in('cardpan', with: "4111111111111111") }
  within_frame(1) { select("1", from: "cardexpiremonth") }
  within_frame(2) { select("2020", from: "cardexpireyear") }
  within_frame(3) { fill_in('cardcvc2', with: "123") } 
  within_frame(4) { select('Visa', from: "cardtype") }

**Comment by twalpole** [comment]
Should be fixed by PR #674 that was merged
